### PR TITLE
Spoiler log improvement, item obtainability validation, bonus fixes, and item description updates.

### LINKED ISSUE #14: Improve spoiler log
- [ ] Remove debug info from items
- [ ] Better descriptions for locations
- [ ] Better organization for locations
- [ ] Add Keyblade names

### LINKED ISSUE #20: Missing bonus levels
- [ ] Hollow Bastion Demyx
- [ ] Additional on Xemnas 1

### PR DESCRIPTION
**Pull request by zlittlefield**
Resolves #14 
Resolves #20 

TL;DR; If I'm overstepping, please feel free to ask me to separate this out, but a lot of these fixes are just QoL improvements for the generator. 

This PR does a few small things that I hope will make things a bit easier for you guys:

1) Makes the spoiler log much more interactive and legible for people making seeds. 
ExampleSpoiler

2) Gives a large number of locations more accurate descriptions to fit the spoiler log (including weapon names, and some more descriptive location names)

3) Fixes the missing bonuses (AFAIK, I've only been able to test so much)

4) Reorganizes the treasure locations by world to aid in future debugging (hopefully)

5) Adds a validation check when making a seed to see if a seed is completable (i.e. all checks are obtainable) and regens a seed if that check fails.

### CODE REVIEW

**Comment by tommadness** [comment]
This is looking great, thank you so much. This is gonna be a huge help going forward!
